### PR TITLE
`Development`: Document the exam timeline

### PR DESCRIPTION
**Pull request by tobias-lippert**
#### General

- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context

The features I added are still missing documentation. This is the first PR to add documentation for one of my features.

### Description
Added docs with screenshots.

### Steps for Testing
Have a look at the code and the docs built in the pipeline.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### CODE REVIEW

**Review by b-fein — APPROVED** [review]

**Review by JohannesStoehr — DISMISSED** [review]
Documentation looks good to me

File: docs/user/exams/instructors_guide.rst [hunk 1]
```diff
@@ -451,10 +451,23 @@ If you want you can also enable the :ref:`second correction <Before the exam's a
 
 - At the bottom of the page you will find the option |check_plagiarism|.
 
+3.4 Exam Timeline
+^^^^^^^^^^^^^^^^^
+- Artemis stores the current state of a submission for text, modeling, and quiz exercises every 30s or whenever the student clicks on the save button.
+- As instructor you have the possibility to view all those states as well as the submissions for file-upload or programming exercises using the exam timeline.
+- The exam timeline is available on the details page of a student exam when the student exam has been submitted by clicking on the button |exam_timeline|. If the exam has not been submitted yet, the exam timeline button is disabled and shows an explanatory tooltip.
+- The exam timeline shows all submissions of the student in chronological order. You can navigate between the different timestamps on the slider or navigating between different exercises using the navigation bar below the timeline
```
Comment (laadvo) [inline docs/user/exams/instructors_guide.rst:459]:
```suggestion
- The exam timeline shows all submissions of the student in chronological order. You can navigate between the different timestamps on the slider or navigate between different exercises using the navigation bar below the timeline
```

File: docs/user/exams/instructors_guide.rst [hunk 2]
```diff
@@ -451,10 +451,23 @@ If you want you can also enable the :ref:`second correction <Before the exam's a
 
 - At the bottom of the page you will find the option |check_plagiarism|.
 
+3.4 Exam Timeline
+^^^^^^^^^^^^^^^^^
+- Artemis stores the current state of a submission for text, modeling, and quiz exercises every 30s or whenever the student clicks on the save button.
```
Comment (laadvo) [inline docs/user/exams/instructors_guide.rst:456]:
```suggestion
- Artemis stores the current state of a submission for text, modeling, and quiz exercises every 30s or whenever the student clicks the save button.
```

File: docs/user/exams/instructors_guide.rst [hunk 3]
```diff
@@ -451,10 +451,23 @@ If you want you can also enable the :ref:`second correction <Before the exam's a
...
+- As instructor you have the possibility to view all those states as well as the submissions for file-upload or programming exercises using the exam timeline.
```
Comment (laadvo) [inline docs/user/exams/instructors_guide.rst:457]:
```suggestion
- As instructor, you have the possibility to view all those states as well as the submissions for file-upload or programming exercises using the exam timeline.
```

File: docs/user/exams/instructors_guide.rst [hunk 4]
```diff
@@ -451,10 +451,23 @@ If you want you can also enable the :ref:`second correction <Before the exam's a
...
+- The exam timeline is available on the details page of a student exam when the student exam has been submitted by clicking on the button |exam_timeline|. If the exam has not been submitted yet, the exam timeline button is disabled and shows an explanatory tooltip.
+- The exam timeline shows all submissions of the student in chronological order. You can navigate between the different timestamps on the slider or navigating between different exercises using the navigation bar below the timeline
+- For programming exercises you can view a git diff between the currently selected submission and the previous submission or the template of this exercise.
```
Comment (laadvo) [inline docs/user/exams/instructors_guide.rst:460]:
```suggestion
- For programming exercises, you can view a git diff between the currently selected submission and the previous submission or the template of this exercise.
```

**Review by laadvo — COMMENTED** [review]
I have some minor suggestions but looks good otherwise :+1:

**Review by laadvo — APPROVED** [review]
thanks for implementing my suggestions, LGTM

**Review by aplr — APPROVED** [review]
Looks good

**Pull request merged**